### PR TITLE
Allow creation of github release tags

### PR DESCRIPTION
**Pull request by adrianmoisey**
##### SUMMARY

Allows a user to create a github release from Ansible.
Useful for anyone doing code releases with Ansible and GitHub Releases.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
github_release
##### ANSIBLE VERSION
```
ansible 2.2.0.0
  config file =
  configured module search path = Default w/o overrides
```

##### ADDITIONAL INFORMATION
A replacement for https://github.com/ansible/ansible/pull/22741 which got stale, and difficult to rebase.

### CODE REVIEW

File: lib/ansible/modules/source_control/github_release.py [hunk 1]
```diff
@@ -53,7 +53,42 @@
         required: true
         description:
             - Action to perform
-        choices: [ 'latest_release' ]
+        choices: [ 'latest_release', 'create_release' ]
+    tag:
+        required: false
+        description:
+            - Tag name when creating a release. Required when using create_release.
```
Comment (Akasurde) [inline lib/ansible/modules/source_control/github_release.py]: Reword as 

`Required when using action is set to C(create_release).`

File: lib/ansible/modules/source_control/github_release.py [hunk 2]
```diff
@@ -145,6 +206,21 @@ def main():
         else:
             module.exit_json(tag=None)
 
+    if action == 'create_release':
+        if tag is None:
+            module.fail_json(msg="Specify tag in order to create a release.")
```
Comment (Akasurde) [inline lib/ansible/modules/source_control/github_release.py]: Use `required_if` 

`required_if = [('action', 'create_release', ['tag'])]`

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/source_control/github_release.py (hunk 2 above)
Comment (adrianmoisey) [reply lib/ansible/modules/source_control/github_release.py]: I've implemented it here: 4399c301a2413f394e78bb048656b9cf744940e9
Please let me know if that's right.

File: lib/ansible/modules/source_control/github_release.py (hunk 1 above)
Comment (adrianmoisey) [reply lib/ansible/modules/source_control/github_release.py]: Fixed in fcdac88cdcb21fa76f0879f3b96d402c8fb6e2dc

**Review by Akasurde — APPROVED** [review]
shipit

**Comment by adrianmoisey** [comment]
:bell:

**Comment by adrianmoisey** [comment]
@Akasurde how do I get this merged?

**Comment by Akasurde** [comment]
@adrianmoisey Try to ping someone on #ansible-devel on IRC, to get review and merge. @gundalow  might help you.

**Pull request merged**